identity/src/keypair.rs

```diff
@@ -67,42 +55,100 @@ use std::convert::{TryFrom, TryInto};
 #[allow(clippy::large_enum_variant)]
 pub enum Keypair {
     /// An Ed25519 keypair.
+    #[cfg(feature = "ed25519")]
+    #[deprecated(
+        since = "0.1.0",
+        note = "This enum will be made opaque in the future, use `Keypair::into_ed25519` instead."
+    )]
```

Comment:
The plan with this is that once users have migrated to the `into_` functions, we can make `Keypair` a struct and have the enum as a private field with `cfg`'d variants. That moves all the code that touches the conditional variants under our control where we can assure that it works under all feature-flag combinations. It will also allow us to remove the `keypair_dummy.rs` file again.